seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -40,29 +46,35 @@
 import java.sql.Connection;
 import java.sql.DatabaseMetaData;
 import java.sql.DriverManager;
+import java.sql.PreparedStatement;
 import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Comparator;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
 public abstract class AbstractJdbcCatalog implements Catalog {
     private static final Logger LOG = LoggerFactory.getLogger(AbstractJdbcCatalog.class);
-
     protected final String catalogName;
     protected final String defaultDatabase;
     protected final String username;
     protected final String pwd;
     protected final String baseUrl;
     protected final String suffix;
     protected final String defaultUrl;
+    protected final JdbcDialect jdbcDialect;
+    protected static final Set<String> SYS_DATABASES = new HashSet<>();
```

Comment:
The static variable bind with `Class`, not `Object`. So in your code, all catalog implement class will share same `SYS_DATABASES`. Please do test before submit code.